### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by cmartinezaraujo**

### CODE REVIEW

**Comment by coproduct** [comment]
• Jester class takes in the correct interfaces: :heavy_check_mark: 
• Jester dependencies are null checked: :heavy_check_mark: 
• Jester appropriately retrieves jokes and sends them to the output: :heavy_check_mark: 
• Chuck Norris jokes get filtered out: :heavy_check_mark: 
• Jester code coverage is above 90%: :heavy_check_mark: 
Extra Credit
• Output to console implementation is unit tested: :x: 
• JokeService uses JSON: :x:

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 1]
```diff
@@ -1,10 +1,73 @@
...
+        public void Jester_TakesValidServices()
+        {
+            JokeService jokeService= new();
+            JokeDisplay jokeDisplay = new();
+
+            Jester joker = new Jester(jokeService, jokeDisplay);
+
+            Assert.AreEqual(joker.JokeService,jokeService);
+            Assert.AreEqual(joker.DisplayService, jokeDisplay);
+        }
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:34]: This works, but a simpler solution is to make the properties private so you don't have to unit test them

File: CanHazFunny/CanHazFunny/IJokeDisplay.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CanHazFunny
+{
+    public interface IJokeDisplay
+    {
+        public void DisplayJoke(string joke);
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny/IJokeDisplay.cs:11]: Avoid specifying access modifiers on interfaces
```suggestion
        void DisplayJoke(string joke);

```

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            _JokeService = jokeService ?? throw new System.ArgumentNullException(nameof(jokeService));
+
+            _DisplayService = tellerService ?? throw new System.ArgumentNullException(nameof(tellerService));
+        }
+
+        public void TellJoke()
+        {
+            string joke = JokeService.GetJoke();
+
+            while(joke.Contains("Chuck") | joke.Contains("Norris"))
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny/Jester.cs:29]: Though this technically works, there is a subtle difference between `|` and `||`. The first is a bitwise or, the second is a logical or. Unless you are looking for bit-wise operations, most of the time you want to use `||`
```suggestion
            while(joke.Contains("Chuck") || joke.Contains("Norris"))

```

**Pull request closed without merging**